dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -207,16 +209,21 @@ public Request.Builder putAllHeaderParams(Multimap<String, ? extends String> ent
         }
 
         public Request.Builder putQueryParams(String key, String... values) {
+            Preconditions.checkNotNull(key, "Query parameter name must not be null");
             mutableQueryParams().putAll(key, Arrays.asList(values));
             return this;
         }
 
         public Request.Builder putQueryParams(String key, String value) {
+            Preconditions.checkNotNull(key, "Query parameter name must not be null");
+            Preconditions.checkNotNull(value, "Query parameter value must not be null");
```

Comment:
Why do we check key+value here, but only key in `public Request.Builder putHeaderParams(String key, String value) {`?

Reply:
headers actually permit null values from what I could determine, while query and path parameters don't

Reply:
if null headers don't result in failures, I'd consider that a bug, but reasonable not to make it throw in this change 👍